src/control/Control.cpp

```diff
@@ -1235,6 +1238,48 @@ void Control::deletePage() {
 
 void Control::insertNewPage(size_t position) { pageBackgroundChangeController->insertNewPage(position); }
 
+void Control::appendNewPdfPages() {
+    int pageCount = this->doc->getPageCount();
+    // find last page with pdf background and get its pdf page number
+    PageRef page;
+    int lastPdfPage = -1;
+    for (int i = pageCount - 1; i >= 0; --i) {
+        page = doc->getPage(i);
+        if (page && page->getBackgroundType().isPdfPage()) {
+            lastPdfPage = page->getPdfPageNr();
+            break;
+        }
+    }
```

Comment:
All of those return values are size_t, so don't cast to int implicitly without bound checks.

```suggestion
    auto pageCount = this->doc->getPageCount();
    // find last page with pdf background and get its pdf page number
    auto currentPdfPageCount = [&]() {
        for (size_t i = pageCount; i != 0; --i) {
            if (auto page = doc->getPage(i - 1); page && page->getBackgroundType().isPdfPage()) {
                return page->getPdfPageNr() + 1;
            }
        }
        return 0ULL;
    }();
```

Reply:
This syntax is new to me. What does the `[&](){ .... }();` stuff do?

Reply:
> This syntax is new to me. What does the `[&](){ .... }();` stuff do?

Oh that's a lamda expression. It is part of the standard since c++11.
Basically I inlined a unnamed function here, returning the value for "currentPdfPageCount" and with the prceding `()` I directly executed it.
With that no predeclared variable is necessary and the returnvalue of the lambda function is directly executed.

https://en.cppreference.com/w/cpp/language/lambda